src/test/java/io/cryostat/net/web/http/api/v2/RulesPostHandlerTest.java

```diff
@@ -220,8 +220,31 @@ void throwsIfOptionalIntegerAttributesNonInteger(String val) {
             MatcherAssert.assertThat(ex.getFailureReason(), Matchers.containsString(val));
         }
 
+        @ParameterizedTest
+        @ValueSource(strings = {"-10", "", "one", "|", "[]"})
+        void throwsIfOptionalJsonAttributesNegativeOrNonInteger(String val) {
+            MultiMap headers = MultiMap.caseInsensitiveMultiMap();
+            Mockito.when(params.getHeaders()).thenReturn(headers);
+            headers.set(HttpHeaders.CONTENT_TYPE, HttpMimeType.JSON.mime());
+
+            String invalidRule =
+                    "{\"name\":\"Invalid_Rule\","
+                            + "\"description\":\"AutoRulesIT automated rule\","
+                            + "\"eventSpecifier\":\"template=Continuous,type=TARGET\","
+                            + "\"targetAlias\":\"es.andrewazor.demo.Main\","
+                            + "\"archivalPeriodSeconds\":"
+                            + val
+                            + "}";
```

Comment:
For easier readability, you could construct this String (and the one like it in the next test too) like:

```
String invalidRule = gson.toJson(Map.of(
  "name": "Invalid_Rule",
  ...
));
```

This way you don't need to deal with manually writing in the JSON formatting with braces, colons, commas, and escaped quotes.

Reply:
You could probably also use `JsonObject` the same way that you do in the `AutoRulesIT` test you added - `toString()` probably does what you need already.